node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/FileFinder.java

```diff
@@ -0,0 +1,251 @@
+            });
+        } catch (NoSuchFileException ignored) {
+
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private static int deleteRecursively(FileAttributes fileAttributes) {
+        int numDeletions = 0;
+        if (fileAttributes.isDirectory()) {
+            numDeletions = FileFinder.from(fileAttributes.path())
+                    .match(all())
+                    .deleteRecursively();
+        }
+
+        return numDeletions + (deleteIfExists(fileAttributes.path()) ? 1 : 0);
+    }
+
+
+    // Ideally, we would reuse the FileAttributes in this package, but unfortunately we only get
+    // BasicFileAttributes and not PosixFileAttributes from FileVisitor
+    public static class FileAttributes {
+        private final Path path;
+        private final BasicFileAttributes attributes;
+
+        FileAttributes(Path path, BasicFileAttributes attributes) {
+            this.path = path;
+            this.attributes = attributes;
+        }
+
+        public Path path() { return path; }
+        public String filename() { return path.getFileName().toString(); }
+        public Instant lastModifiedTime() { return attributes.lastModifiedTime().toInstant(); }
+        public boolean isRegularFile() { return attributes.isRegularFile(); }
+        public boolean isDirectory() { return attributes.isDirectory(); }
+        public long size() { return attributes.size(); }
+    }
+
+
+    // Filters
+    public static Predicate<FileAttributes> olderThan(Duration duration) {
+        return attrs -> Duration.between(attrs.lastModifiedTime(), Instant.now()).compareTo(duration) > 0;
+    }
+
+    public static Predicate<FileAttributes> youngerThan(Duration duration) {
+        return olderThan(duration).negate();
+    }
+
+    public static Predicate<FileAttributes> largerThan(long sizeInBytes) {
+        return attrs -> attrs.size() > sizeInBytes;
+    }
+
+    public static Predicate<FileAttributes> smallerThan(long sizeInBytes) {
+        return largerThan(sizeInBytes).negate();
+    }
+
+    public static Predicate<FileAttributes> nameMatches(Pattern pattern) {
+        return attrs -> pattern.matcher(attrs.filename()).matches();
+    }
+
+    public static Predicate<FileAttributes> nameStartsWith(String string) {
+        return attrs -> attrs.filename().startsWith(string);
+    }
+
+    public static Predicate<FileAttributes> nameEndsWith(String string) {
+        return attrs -> attrs.filename().endsWith(string);
+    }
+
+    public static Predicate<FileAttributes> all() {
+        return attrs -> true;
+    }
+
+
+    // Other helpful methods that no not throw checked exceptions
+    public static boolean moveIfExists(Path from, Path to) {
+        try {
+            Files.move(from, to);
+            return true;
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public static boolean deleteIfExists(Path path) {
```

Comment:
I think these would be better suited for UnixPath than FileFinder

Reply:
Moved.